src/main/kotlin/it/unibo/protelis/web/execution/simulated/SimulatedProtelisEngine.kt

```diff
@@ -3,77 +3,69 @@ package it.unibo.protelis.web.execution.simulated
 import it.unibo.alchemist.core.implementations.Engine
 import it.unibo.alchemist.core.interfaces.Simulation
 import it.unibo.alchemist.core.interfaces.Status
-import it.unibo.alchemist.core.interfaces.awaitFor
+import it.unibo.alchemist.core.interfaces.waitForAndCheck
 import it.unibo.alchemist.loader.YamlLoader
 import it.unibo.alchemist.model.implementations.positions.Euclidean2DPosition
 import it.unibo.alchemist.model.implementations.times.DoubleTime
-import it.unibo.protelis.web.execution.CoroutineProtelisEngine
 import it.unibo.protelis.web.execution.ProtelisObserver
-import java.util.concurrent.atomic.AtomicReference
-import kotlinx.coroutines.Deferred
 import kotlinx.coroutines.GlobalScope
-import kotlinx.coroutines.async
-import kotlinx.coroutines.future.asCompletableFuture
 import kotlinx.coroutines.launch
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 
-class SimulatedProtelisEngine() : CoroutineProtelisEngine() {
+class SimulatedProtelisEngine {
   private val logger: Logger = LoggerFactory.getLogger(this::class.java)
-  private var alchemistEngine: AtomicReference<Simulation<Any /* TODO */, Euclidean2DPosition>?> = AtomicReference(null)
+  private var alchemistEngine: Simulation<Any, Euclidean2DPosition>? = null
 
-  init {
-    logger.debug("Simulation container created")
-  }
-
-  constructor(sourceCode: String, monitor: ProtelisObserver) : this() {
-    GlobalScope.launch { setupAwait(sourceCode, monitor) }.asCompletableFuture().join()
-  }
+  fun setup(sourceCode: String, monitor: ProtelisObserver) {
+    if (alchemistEngine != null) {
+      stop()
+    }
 
-  override suspend fun setupAwait(sourceCode: String, monitor: ProtelisObserver) {
-    if (alchemistEngine.get() != null) stopAwait()
-    val sim = setupSimulationAsync().await()
-    alchemistEngine.set(sim)
+    alchemistEngine = setupSimulation()
     setCode(sourceCode)
-    alchemistEngine.get()
+    alchemistEngine
       ?.addOutputMonitor(ProtelisUpdateAdapter(monitor))
       ?: throw IllegalStateException("Simulation was not set up correctly")
-    alchemistEngine.get()
-      ?.awaitFor(Status.INIT)
+    alchemistEngine
+      ?.waitForAndCheck(Status.INIT)
       ?: throw IllegalStateException("Simulation was not set up correctly")
-    GlobalScope.launch { sim.run() }
+    GlobalScope.launch() { alchemistEngine?.run() ?: throw IllegalStateException() }
     logger.debug("Simulation Engine set up correctly")
   }
 
-  override suspend fun startAwait() {
-    checkNotNull(alchemistEngine.get())
-    alchemistEngine.get()?.play()
-    alchemistEngine.get()?.awaitFor(Status.RUNNING) // TODO: should I wait RUNNING instead?
+  fun start() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [UndocumentedPublicFunction - [start] at /src/src/main/kotlin/it/unibo/protelis/web/execution/simulated/SimulatedProtelisEngine.kt:37:3](https://app.codacy.com/manual/NiccoMlt/Protelis-Web/pullRequest?prid=5040950)